src/puppetlabs/pcp/message.clj

```diff
@@ -17,6 +18,14 @@
   ;; accept that we can have anything in memory, but we'll check the
   ;; Envelope schema when interacting with the network
   (merge Envelope
+         {:sender s/Str}))
```

Comment:
it's still required for a test. I've left it in place for now, but I think there are probably a number of tests we might remove from this repo on closer review.